provider/cmd/pulumi-gen-eks/main.go

```diff
@@ -397,6 +397,26 @@ func generateSchema() schema.PackageSpec {
 						TypeSpec:    schema.TypeSpec{Type: "integer"},
 						Description: "The size in GiB of a cluster node's root volume. Defaults to 20.",
 					},
+					"nodeRootVolumeDeleteOnTermination": {
+						TypeSpec:    schema.TypeSpec{Type: "boolean"},
+						Description: "Whether to delete a cluster node's root volume on termination. Defaults to true.",
```

Comment:
You can set the `Default` parameter to encode the default value in the schema. This should generate the appropriate default handling code for you.